### DISCUSSION TITLE
PulseTime for more than 8 relays

### DISCUSSION THREAD

**Discussion by tsunfinger** [arendst/Tasmota#14825] [Q&A] [answered]
I have running ESP32  on 10.1.0 with 9 relays, but PulseTime only counts up to 8. Regarding documentation I have on relay 9 the same PulseTime as on relay 1.
This is not the way I can use it. Is there a workaround or any plans to implement PulseTime 9-16?

Cheers  Tom

**Comment by barbudor** [comment]
Sorry for my 1st answer, I thought you said PulseTime was not working as per the documentation.

There is no way to have more than 8 Pulsetime values.
You may want to use RulesTimer

But on ESP32 the simplest is to use Berry and have as many timers as you want

**Reply by tsunfinger** [reply]
My Tasmota devices are controlled by MQTT so Berry doesn't look like an option for me.

**Comment by barbudor** [comment]
Why ?
Have you really explored Berry ?
You can have rules and as many timers you want

**Reply by tsunfinger** [reply]
and I can set timer by MQTT?

**Reply by barbudor** [reply]
You can create your own command in Berry
So you can create one to set the timer value

**Reply by tsunfinger** [reply]
The timer value has to be set by my home automation with MQTT. Berry doesn't sound like a solution for missing PulseTime 9-16 for my setup.

**Reply by sfromis** [reply]
Berry can be just fine for controlling relay actions, and whatever timing you want, including easily creating control commands available via MQTT. Of course, you'd have to do a bit of programming for that

**Comment by arendst** [comment, marked as answer]
32 pulstimers are supported in v11.0.0.3 released today
